src/elexclarity/formatters/results.py

```diff
@@ -101,7 +128,58 @@ def format_top_level_counts(self, choices):
 
         return counts
 
-    def format_race(self, contest, election_date, election_type, timestamp=None, level=None, county_id=None, **kwargs):
+    def _get_precinct_fully_reporting_statuses_via_percent_reporting(self, election):
+        """
+        Constructs a mapping from precinct IDs to boolean flags representing
+        whether or not the given precinct is fully reporting. This method does so by looking
+        at the VoterTurnout from the ElectionResults tag in the precinct details xml
+        and checking if ``percentReporting`` is 4. This means (we think) that all 4 vote
+        types (early/absentee/day of in person/provisional) have been reported for the
+        given precinct.
+        """
+        subunit_fully_reporting_statuses = {}
+        precincts = get_list(election["VoterTurnout"]["Precincts"]["Precinct"])
+        for precinct in precincts:
+            subunit_id = self.get_precinct_id(precinct["name"])
+            is_precinct_fully_reporting = (precinct.get("percentReporting") == str(self.NUM_VOTE_TYPES))
+            subunit_fully_reporting_statuses[subunit_id] = is_precinct_fully_reporting
+
+        return subunit_fully_reporting_statuses
+
+    def _get_precinct_fully_reporting_statuses_via_vote_types(self, contest):
+        """
+        Constructs a mapping from precinct IDs to boolean flags representing
+        whether or not the given precinct is fully reporting. This method does so by looking
+        at the VoterTurnout from the ElectionResults tag in the precinct details xml
+        and checking if ``percentReporting`` is 4. This means (we think) that all 4 vote
+        types (early/absentee/day of in person/provisional) have been reported for the
+        given precinct.
+        """
```

Comment:
Tiny detail, but it looks like this docstring might be copy-pasted from the one above?